### PR TITLE
Support refresh_token in Ruby Library

### PR DESCRIPTION
**Pull request by alexandrebini**
refresh

## What?

Support refresh_token in Ruby Library

## Why?

Because by default, our access token expires in 7200 seconds and users need a ease way to handle it

## How to test?

```ruby
require 'redbooth-ruby'

RedboothRuby.config do |configuration|
  configuration[:consumer_key] = '_your_consumer_key_'
  configuration[:consumer_secret] = '_your_consumer_secret_'
end

session = RedboothRuby::Session.new(
  token: '_your_user_token_',
  refresh_token: '_your_user_refresh_token_',
  auto_refresh_token: true,
  on_token_refresh: Proc.new do |old_access_token, new_access_token|
    auth = Authentication.where(access_token: old_access_token.token).first
    auth.access_token = new_access_token.token
    auth.refresh_token = new_access_token.refresh_token
    auth.save
  end
)

client = RedboothRuby::Client.new(session)
p client.me(:show)
```

### CODE REVIEW

**Comment by botdylan** [comment]
&nbsp;<b>Code Climate</b> has <a href="https://codeclimate.com/repos/5461c4f6e30ba075bc0a0ab0/compare/feature/refresh_token">analyzed this pull request</a>.

File: README.md [hunk 1]
```diff
@@ -16,23 +16,23 @@ Usage
 First, you've to install the gem
 
 ```Ruby
-  gem install redbooth-ruby
```
Comment (alexandrebini) [inline README.md]: I just adjusted the indentation of the code block... sorry OCD :warning:

**Comment by bpedro** [comment]
:+1:

File: lib/redbooth-ruby/request/validator.rb [hunk 2]
```diff
@@ -47,9 +47,9 @@ def verify_response_code
       def verify_authentication_header
         case raw_response.headers['WWW-Authenticate']
         when /error\=\"invalid_token\".*expired/
-          fail OauhtTokenExpired
+          fail OauthTokenExpired
```
Comment (sauloperez) [inline lib/redbooth-ruby/request/validator.rb]: lol

File: lib/redbooth-ruby/session.rb [hunk 3]
```diff
@@ -31,16 +36,34 @@ def client
...
 
     def get_access_token_url
-      url = OAUTH_URLS[:request_token_url]
-      url += "?oauth_verifier=#{oauth_verifier}" if oauth_verifier
-      url += "&oauth_token=#{oauth_token}" if oauth_token
+      uri = URI.parse(OAUTH_URLS[:token_url])
+      params = URI.decode_www_form(uri.query.to_s)
+      params << ['oauth_verifier', oauth_verifier] if oauth_verifier
+      params << ['oauth_token', oauth_token] if oauth_token
+      uri.query = URI.encode_www_form(params) if params.size > 0
```
Comment (sauloperez) [inline lib/redbooth-ruby/session.rb]: :clap:

Comment (alexandrebini) [inline lib/redbooth-ruby/session.rb]: :sweat_smile:

File: lib/redbooth-ruby/client_operations/perform.rb [hunk 4]
```diff
@@ -0,0 +1,89 @@
+module RedboothRuby
```
Comment (alexandrebini) [inline lib/redbooth-ruby/client_operations/perform.rb]: @masylum No more issues on Code Climate, tks! :+1: 

https://codeclimate.com/repos/5461c4f6e30ba075bc0a0ab0/compare/feature/refresh_token

File: spec/cassettes/RedboothRuby_Project/_delete/makes_a_new_DELETE_request_using_the_correct_API_endpoint_to_delete_a_specific_project.yml [hunk 5]
```diff
@@ -121,8 +121,8 @@ http_interactions:
       - thin 1.6.1 codename Death Proof
     body:
       encoding: UTF-8
-      string: '{"status":"Enqueued","process_token":null,"retry_after":20}'
-    http_version: 
+      string: '{"status":"Enqueued","process_token":null,"retry_after":0}'
```
Comment (alexandrebini) [inline spec/cassettes/RedboothRuby_Project/_delete/makes_a_new_DELETE_request_using_the_correct_API_endpoint_to_delete_a_specific_project.yml]: Just removed all this retry_after delays... they were slowing the tests

**Comment by andresgutgon** [comment]
@alexandrebini I think in our ruby is with spaces `"#{ foo }"` Right @csaura ?

**Comment by andresgutgon** [comment]
:+1: remove `WIP` if finished

**Comment by alexandrebini** [comment]
this library is using always ```"#{foo}"```. I can change if you want.

**Comment by alexandrebini** [comment]
@andresgutgon done!

**Comment by jrom** [comment]
Awesome! :+1:

**Pull request merged**